.github/CONTRIBUTING.md

```diff
@@ -167,50 +167,55 @@ Additional references:
+   - If the build fails, a [Repository Maintainer][repository-maintainer] adds the `Review - waiting on author` label to the pull request. 
+   The *author* can then continue to update the pull request until the build passes.
+1. If the *author* knows whom should participate in the review, they should add them otherwise they can add the recommended *reviewers*.
+1. Once the build passes, if there is not sufficient review, the *maintainer* adds the `Review - needed` label.
+1. An [Area Expert][area-expert] should also review the pull request.
+   - If the *author* does not meet the *reviewer*'s standards, the *reviewer* makes comments. A *maintainer* then removes the `Review - needed` label and adds 
+   the `Review - waiting on author` label. The *author* must address the comments and repeat from step 2.
+   - If the *author* meets the *reviewer*'s standards, the *reviewer* approves the PR. A maintainer then removes the `need review` label.
+1. Once the code review is completed, a *maintainer* merges the pull request after one business day to allow for additional critical feedback.
+
+#### Pull Request - Roles and Responsibilities
+
+1. The PR *author* is responsible for moving the PR forward to get it Approved.
+   This includes addressing feedback within a timely period and indicating feedback has been addressed by adding a comment and mentioning the specific *reviewers*.
+   When updating your pull request, please **create new commits** and **don't rewrite the commits history**. 
+   This way it's very easy for the reviewers to see diff between iterations.
+   If you rewrite the history in the pull request, review could be much slower.
+   The PR is likely to be squashed on merge to master by the *assignee*.
+1. *Reviewers* are anyone who wants to contribute.
+   They are responsible for ensuring the code: addresses the issue being fixed, does not create new issues (functional, performance, reliability, or security), and implements proper design.
+   *Reviewers* should use the `Review changes` drop down to indicate they are done with their review.
+   - `Request changes` if you believe the PR merge should be blocked if your feedback is not addressed,
+   - `Approve` if you believe your feedback has been addressed or the code is fine as-is, it is customary (although not required) to leave a simple "Looks good to me" (or "LGTM") as the comment for approval.
+   - `Comment` if you are making suggestions that the *author* does not have to accept.
+   Early in the review, it is acceptable to provide feedback on coding style based on the published [Coding Guidelines](../docs/dev-process/coding-guidelines.md), however,
+   after the PR has been approved, it is generally _not_ acceptable to focus on style issues as it disrupts the PR process.
+   Non-critical late feedback (after PR has been approved) can be submitted as a new issue or new pull request from the *reviewer*.
```

Comment:
I repeat my comment (it seems it may be lost):

I agree with @daxian-dbw. What is "style issues"? If the commits contains bad code, bad patterns and broken formattings we should request obligatory changes in any time. Any policy that allow pass bad PR is bad policy.
 Currently we fast merge with ignoring minor issues only "blocking" and "critical" PRs. Also I think we can fast merge if author ask us (busy, vacation...) but before that, we have to open the tracking Issue and determine (PR Assignee) exactly who will do fix it as soon as possible. In fact, we are doing so.

Reply:
The intent here (if it's not captured well, let me know how to fix) is that we welcome style issues along with other types of issues, but if the only remaining problems after a PR has been approved, it's _generally_ not acceptable to provide only style issues.  In this situation, if the issue is deemed blocking, then go ahead and submit the feedback.  If not and a maintainer, go ahead and make the changes directly.  If not a maintainer, open a new `code cleanup` issue or submit a follow-up PR.

Reply:
I am 11 months in the project and trying look every PR - I never see any problems with style change requests - all authors always address such requests quickly. Moreover we have many PR in which last commits is just fix style issues - extra lines, spaces, unneeded and obvious comments, commented code and so on. It's a really minute job, but we're getting a clean code.
If we're going to skip the dirty code who's going to peel it? Assignee who pass it?

Reply:
@iSazonov I appreciate all the work and impact you've provided, so try not to take this one personally as although this update coincides with a recent PR, I've received similar feedback from others.  I'm hoping that your metatests along with potentially some other tooling suggestions @TravisEz13 has made internally, we don't have to worry as much about style and focus on the code.  

A bigger part of this doc change is getting people to use the GitHub comment system to consistently indicate whether they are done with the review, and if the feedback is blocking or not.

Reply:
For the first two months of this project, I was very amazed at the demands of maintainers for strict style. Later, when I had the experience of writing code for the project, I realized that it was not only justified, but necessary - following a good style saves our time and protects us from making mistakes.
In the project we can see samples of good style code and bad style code. I'm sure everyone would prefer to work with good style code to keep their strength and their time, but at the same time some people you say don't want to create good style code. I don't understand why we should replace good policy with bad policy. I believe MSFT should provide the best practices, especially in public projects, that are good for her reputation.

Reply:
I gave it some more thoughts, and think that we probably can tweak the words a bit. For example, what if the first reviewer approved the PR very quickly? Then does that mean the subsequent reviewers cannot comment on code style issue (based on coding guidelines) anymore?

How about changing it to this? `"however, after the PR has been approved, it is generally not recommended to focus on style issues unless they go against the Coding Guidelines"`

Reply:
And similar to any other feedbacks, for review iteration that focuses on code styles, if the reviewer thinks some style issues block the PR, then the reviewer should select `request for changes` for that feedback and explain why that is a blocking issue (referring to the Coding Guidelines). For style issues that don't block the PR, the reviewer can select `comments` for that feedback.

I don't differentiate `comments` and `request changes` that much in the past (both as a reviewer and a maintainer), but will pay more attention in future.

Reply:
I'm fine with the language change recommendation from @daxian-dbw and will update.  Ultimately, I'm trying to find a balance between maintaining high quality and improving efficiency.  I believe some additional tooling will help make this easier for both the contributor/author, reviewers, and maintainers.

Reply:
And for the word `code style issue`, maybe it's too general because I feel what @iSazonov has in mind is issues like repeated codes (violating the DRY rule) and other things that actually are wrong coding patterns. For those, we definitely want the feedback to be addressed.

Maybe we should be more specific and use the word `code format issue`? 
Below are the formatting issue examples listed in 'Coding Guidelines', and I believe they are more in line with what we mean here.

>A basic rule of formatting is to use "Visual Studio defaults".
Here are some general guidelines
>* No tabs, indent 4 spaces.
>* Braces usually go on their own line, with the exception of single line statements that are properly indented.
>* Use `_camelCase` for instance fields, use `readonly` where possible.
>* Use of `this` is neither encouraged nor discouraged.
>* Avoid more than one blank empty line.
>* Public members should use [doc comments](https://msdn.microsoft.com/en-us/library/b2s063f7.aspx), internal members may use doc comments but it is not encouraged.
>* Public members in a namespace that ends with `Internal`, for example `System.Management.Automation.Internal` are not considered a supported public API. Such members are necessarily public as implementation details in code shared between C# and PowerShell script, or must be available publicly by generated code.
>* File encoding should be ASCII (preferred) or UTF8 (with BOM) if absolutely necessary.

Reply:
BTW, I will submit a PR soon to add additional things to the "Coding Guidelines" doc. We can together make that doc a better reference to use during review time.

Reply:
For me style == formatting only, but I can see how it might be interpreted differently.  I'm fine changing it to formatting since that's what we use in the coding guidelines.  Agree that non-formatting feedback should always be given although even in those cases, some can be taken as a different issue.

Reply:
I see different understanding the term "style" too. 

@daxian-dbw It is good idea to enhance the "Coding Guidelines". We should think about sync our Meta tests with "Coding Guidelines" doc and how activate the tests.

Reply:
I absolutely agree.